### PR TITLE
[NODE-TRAINING] Sign in

### PR DESCRIPTION
**Pull request by alanlamaswolox**
## Summary
Functionality to sign in

## Trello Card
[link](https://trello.com/c/jDIfBEKE/3-sign-in)

### CODE REVIEW

File: app/controllers/users.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 const logger = require('../logger');
 const usersService = require('../services/users');
+const credentialsHelper = require('../helpers/users/validate_credentials');
```
Comment (SantiagoFoster) [inline app/controllers/users.js]: Remember to name files in camelCase. Also, a function to handle the controllers logic belongs in an interactor or service, not a helper, specially since it access the db.

File: test/controllers/user.test.js [hunk 2]
```diff
@@ -66,3 +67,52 @@ describe('usersController.signUp', () => {
       });
   });
 });
+
+describe('usersController.signIn', () => {
+  it('Log in with previously created user', () => {
+    userAttributes.password = 'password1923';
+    return request
+      .post('/users')
```
Comment (SantiagoFoster) [inline test/controllers/user.test.js]: Have a look at this: https://www.npmjs.com/package/factory-girl for generating users. It won't add much now but it really helps when you need to generate several models.

Comment (SantiagoFoster) [inline test/controllers/user.test.js]: Also, user creation is a task shared between all 3 test and should be in a `beforeEach`: https://jestjs.io/docs/en/setup-teardown

File: app/controllers/users.js [hunk 3]
```diff
@@ -9,3 +10,11 @@ module.exports.signUp = (req, res, next) =>
...
     })
     .catch(next);
+
+module.exports.signIn = (req, res, next) =>
+  credentialsHelper
+    .signIn(req.body)
+    .then(token => {
+      res.status(200).send({ response: token });
+    })
+    .catch(next);
```
Comment (SantiagoFoster) [inline app/controllers/users.js]: All endpoints should have at leats one log detailing what it is doing and showing the param that identifies the main resource. 

Also, something I've just realized (and talked with a TL so it can be added to future trainings):
We have a logger middleware that automatically logs the body of the request (it is deactivated while testing, but if you run the server and call an endpoint you can see it).
Said logs are dangerous because they show sensitive information (passwords in this case). Please configure said middleware to censor the bodies of the signUp and signIn enpoints, as shown here:
https://www.npmjs.com/package/express-wolox-logger#obfuscating-body-of-specific-request

**Review by SantiagoFoster — CHANGES_REQUESTED** [review]
Good work! Just a few observations

File: test/controllers/user.test.js (hunk 2 above)
Comment (alanlamaswolox) [reply test/controllers/user.test.js]: I prefer let this like one method than register a specific user, in the next PR i'll add and use that package.

File: test/controllers/user.test.js [hunk 4]
```diff
@@ -66,3 +69,35 @@ describe('usersController.signUp', () => {
       });
   });
 });
+
+describe('usersController.screateUserignIn', () => {
+  beforeEach(() => {
+    userAttributes.email = 'omar.rodriguez@wolox.com';
+    userAttributes.password = 'password1923';
+    return createUser();
```
Comment (SantiagoFoster) [inline test/controllers/user.test.js]: You should not pass params to a function by editing an existing object. This is super hard to follow for other people looking at your code.
`createUser` should either receive both email and pass or an object containing them.

File: app/controllers/users.js (hunk 3 above)
Comment (SantiagoFoster) [reply app/controllers/users.js]: You are missing a log in the controller still:
All endpoints should have at leats one log detailing what it is doing and showing the value of the param that identifies the main resource, in this case the email.

**Review by SantiagoFoster — CHANGES_REQUESTED** [review]

File: test/controllers/user.test.js [hunk 5]
```diff
@@ -4,20 +4,21 @@ const models = require('../../app/models/index');
...
-const createUser = () => request.post('/users').send(userAttributes);
+const createUser = (firstName, lastName, email, password) =>
+  request.post('/users').send(userAttributes(firstName, lastName, email, password));
 
 describe('usersController.signUp', () => {
   it('Creates a user', () =>
     request
       .post('/users')
-      .send(userAttributes)
+      .send(userAttributes('Omar', 'Rodriguez', 'omar.rodriguez@wolox.com', 'password1923'))
```
Comment (SantiagoFoster) [inline test/controllers/user.test.js]: You could have reused createUser here. Ditto for the rest of the tests.

**Review by SantiagoFoster — CHANGES_REQUESTED** [review]

**Review by SantiagoFoster — APPROVED** [review]
Please squash your commits
